consensus/service/src/config.rs

```diff
@@ -94,6 +107,26 @@ fn keypair_from_base64(private_key: &str) -> Result<Arc<Ed25519Pair>, String> {
     Ok(Arc::new(Ed25519Pair::from(secret_key)))
 }
 
+/// Converts a hex-encoded string into an array of 32 bytes.
+fn from_hex_32(src: &str) -> Result<[u8; 32], String> {
+    let bytes = hex::decode(src).map_err(|err| format!("Invalid input: {}", err))?;
```

Comment:
Recommend check `src.len()` and using [`hex::decode_to_slice()`](https://docs.rs/hex/0.4.2/hex/fn.decode_to_slice.html) instead.